### PR TITLE
Implemented Ad Progress detector

### PR DESCRIPTION
**Pull request by balavor**
## Changes
- Implemented Ad Progress detector.
## Workflow tests
screen shot 2019-02-13 at 19 54 25

@VerizonAdPlatforms/video-partner-sdk-developers: Please review.

### CODE REVIEW

File: sources/metrics/detectors/ad progress/AdProgressDetector.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import Foundation
+import PlayerCore
+
+extension Detectors {
+    final class AdProgressDetector {
+        
+        private var processed = Set<UUID>()
+        
+        func process(state: PlayerCore.State) -> Bool {
+            return false
```
Comment (VladyslavAnokhin) [inline sources/metrics/detectors/ad progress/AdProgressDetector.swift]: Nice WT screenshots :)

File: sources/metrics/tracking pixels/TrackingPixelsConnector_Ad.swift [hunk 2]
```diff
@@ -305,6 +305,16 @@ extension TrackingPixels.Connector {
                 }
             }
         }
+        /*Ad Progress Detector*/ do {
+            state.vrmFinalResult.successResult?.inlineVAST.pixels.progress.forEach {
```
Comment (AndriiMoskvin) [inline sources/metrics/tracking pixels/TrackingPixelsConnector_Ad.swift]: WTF - please put this inside detector.

**Review by AndriiMoskvin — DISMISSED** [review]

File: sources/metrics/detectors/ad progress/AdProgressDetectorTest.swift [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    
+    override func setUp() {
+        super.setUp()
+        
+        sut = Detectors.AdProgressDetector()
+        timeProgress = .init(url: testUrl, offset: .time(30))
+        percentageProgress = .init(url: testUrl, offset: .percentage(50))
+    }
+    
+    func testProcessNoOffset() {
```
Comment (VladyslavAnokhin) [inline sources/metrics/detectors/ad progress/AdProgressDetectorTest.swift]: Also, nice to have a test case for detecting same time

File: sources/utils/Utils.swift [hunk 4]
```diff
@@ -73,3 +73,5 @@ func ==(lhs: [[JSON]], rhs: [[JSON]]) -> Bool {
     }
     fatalError("Incorrect equality operator implementation!")
 }
+
+let testUrl = URL(string: "http://test.com")!
```
Comment (VladyslavAnokhin) [inline sources/utils/Utils.swift]: bad idea to add a global variable even for test. You can override already existing data

Comment (balavor) [reply sources/utils/Utils.swift]: Check player core.

File: sources/metrics/detectors/ad progress/AdProgressDetectorTest.swift (hunk 3 above)
Comment (balavor) [reply sources/metrics/detectors/ad progress/AdProgressDetectorTest.swift]: Same time?

Comment (VladyslavAnokhin) [reply sources/metrics/detectors/ad progress/AdProgressDetectorTest.swift]: if we get current time same as we've got on the previous step

Comment (balavor) [reply sources/metrics/detectors/ad progress/AdProgressDetectorTest.swift]: Well, we have similar logic - we get a value that is higher than offset. I use `>=`, not `==`. Anyway, I'm gonna move this logic in core

**Comment by balavor** [comment]
PR is updated and RFR.

File: sources/metrics/detectors/ad progress/AdProgressDetector.swift [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        
+        func process(currentTime: Double,
+                     progressPixelsArray: [PlayerCore.AdProgress.Pixel]) -> [URL] {
+            guard progressPixelsArray.isEmpty == false else {
+                offsetProcessed = []
+                return []
+            }
+            return progressPixelsArray
+                .filter {
+                    return $0.offsetInSeconds == Int(currentTime.rounded()) &&
```
Comment (VladyslavAnokhin) [inline sources/metrics/detectors/ad progress/AdProgressDetector.swift]: no need to use `return`

File: sources/metrics/detectors/ad progress/AdProgressDetector.swift [hunk 6]
```diff
@@ -0,0 +1,29 @@
+//  Copyright 2019, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.hts reserved.
+
+import Foundation
+import PlayerCore
+
+extension Detectors {
+    final class AdProgressDetector {
```
Comment (VladyslavAnokhin) [inline sources/metrics/detectors/ad progress/AdProgressDetector.swift]: `AdVASTProgressDetector` ?

**Review by VladyslavAnokhin — DISMISSED** [review]

File: sources/metrics/detectors/ad progress/AdProgressDetector.swift (hunk 6 above)
Comment (RomanTysiachnik) [reply sources/metrics/detectors/ad progress/AdProgressDetector.swift]: Not vast anymore

File: sources/metrics/detectors/ad progress/AdProgressDetector.swift (hunk 5 above)
Comment (balavor) [reply sources/metrics/detectors/ad progress/AdProgressDetector.swift]: Right

**Review by VladyslavAnokhin — APPROVED** [review]

**Pull request merged**